### PR TITLE
IDNECHO-2425 - Add feature to validate CC V1 API calls

### PR DESCRIPTION
**Pull request by diego-desani-sp**
### Changes
- Adding features and utility classes to validate CC V1 API calls.
- Refactoring names and file location.
- Adding username and password to Karate config. These values are used to validate CC V1 API calls.

### E2E tests
- Green: https://construct.identitysoon.com/view/Styx/job/styx-verify/10/

### CODE REVIEW

File: src/main/java/com/sailpoint/util/AuthUtil.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	 * @param username the username
+	 * @param password the password
+	 * @return the SHA-256-based hash
+	 */
+	public static String getUsernamePasswordHash(String username, String password) {
+		String hash = "";
+		try {
+			hash = encodeString(password + encodeString(username));
+		} catch(Exception ex){
+			ex.printStackTrace();
```
Comment (william-gooch-sp) [inline src/main/java/com/sailpoint/util/AuthUtil.java]: Log something here, printing the stack trace isn't helpful.

File: src/main/java/com/sailpoint/util/AuthUtil.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	 * @return the SHA-256 hash, hex-encoded
+	 */
+	public static String encodeString(String string) throws Exception{
+		String returnString;
+		try {
+			MessageDigest md = MessageDigest.getInstance("SHA-256");
+			md.update(string.getBytes("UTF-8"));
+			returnString = new String(Hex.encode(md.digest()));
+		} catch (Exception ex) {
+			ex.printStackTrace();
```
Comment (william-gooch-sp) [inline src/main/java/com/sailpoint/util/AuthUtil.java]: Don't print the stack trace, log instead.  Why catch and rethrow the same Exception?

Comment (diego-desani-sp) [reply src/main/java/com/sailpoint/util/AuthUtil.java]: This helper code comes from: https://github.com/sailpoint/identitynow-sdk/blob/ebf15dd4384788adeeda999174a7cdc6b1889c5c/src/main/java/sailpoint/cloud/sdk/utility/PasswordUtil.java

But agree with you.

**Comment by diego-desani-sp** [comment]
New test run after changes:
- https://construct.identitysoon.com/view/Styx/job/styx-verify/12/

File: src/main/java/com/sailpoint/util/AuthUtil.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	 * Computes the SHA-256 hash of the specified string, returned in
+	 * hex-encoded format.
+	 *
+	 * @param string the string
+	 * @return the SHA-256 hash, hex-encoded
+	 */
+	private static String encodeString(String string) {
+		try {
+			MessageDigest md = MessageDigest.getInstance("SHA-256");
+			md.update(string.getBytes("UTF-8"));
```
Comment (william-gooch-sp) [inline src/main/java/com/sailpoint/util/AuthUtil.java]: Hmm, might want the encoding to be settable by the caller.  Not a show-stopper though, generally UTF-8 is good.

**Pull request merged**